### PR TITLE
feat: add typeahead support for Datadog metric config

### PR DESCRIPTION
**Pull request by maggieneterval**
Adds typeahead support for Datadog metric configuration, based on existing logic for Prometheus and Stackdriver.

@dotdotdotpaul Would you mind verifying that this works as expected with your Datadog setup? Thank you so much!

datadogtypeahead

### CODE REVIEW

**Comment by dotdotdotpaul** [comment]
I'll try to get to this sometime today.

File: src/kayenta/metricStore/datadog/metricConfigurer.tsx [hunk 1]
```diff
@@ -24,27 +24,17 @@ export const queryFinder = (metric: ICanaryMetricConfig) => get(metric, 'query.m
...
-            onChange={this.onChange}
-            disabledStateKeys={[DISABLE_EDIT_CONFIG]}
-          />
-        </FormRow>
-      </section>
-    );
-  }
+function DatadogMetricConfigurer({ changeMetricName, editingMetric }: DatadogMetricConfigurerProps) {
+  return (
+    <section>
```
Comment (danielpeach) [inline src/kayenta/metricStore/datadog/metricConfigurer.tsx]: This tag could probably be dropped?

File: src/kayenta/actions/index.ts [hunk 2]
```diff
@@ -45,6 +45,7 @@ export const UPDATE_SCORE_THRESHOLDS = 'update_score_thresholds';
 export const UPDATE_GROUP_WEIGHT = 'update_group_weight';
 export const SET_CONFIG_JSON_MODAL_TAB_STATE = 'set_config_json_modal_tab_state';
 export const UPDATE_ATLAS_QUERY = 'update-atlas-query';
+export const UPDATE_DATADOG_METRIC_TYPE = 'update-datadog-metric-type';
```
Comment (danielpeach) [inline src/kayenta/actions/index.ts]: It looks like the delimiter here has been inconsistent - could you please switch this one and the others to use an underscore?

File: src/kayenta/metricStore/datadog/metricConfigurer.tsx (hunk 1 above)
Comment (maggieneterval) [reply src/kayenta/metricStore/datadog/metricConfigurer.tsx]: Updated!

File: src/kayenta/actions/index.ts (hunk 2 above)
Comment (maggieneterval) [reply src/kayenta/actions/index.ts]: Good catch, updated

**Comment by dotdotdotpaul** [comment]
I'm not going to be able to get around to playing with this for at least a few days -- if you're okay waiting, I'll try to get to it when I can, otherwise, seems like a no-harm-no-foul kind of commit.

**Comment by maggieneterval** [comment]
Hey @dotdotdotpaul -- I think we're okay waiting a few days, really appreciate you checking this out when you have time.

**Comment by dotdotdotpaul** [comment]
Is there an update to the kayenta backend that goes with this?  No matter what I type in, I'm getting an empty set back.  Could be a bad configuration on my side, or an odd incompatibility because the backend I'm testing against is older.

**Comment by maggieneterval** [comment]
@dotdotdotpaul Sorry, should have let you know that @duftler added the backend component for this a little over a week ago with [this release](https://github.com/spinnaker/kayenta/releases/tag/v1.0.21) (and more specifically, [this commit](https://github.com/spinnaker/kayenta/commit/491dd093831ab25feaf778efa5ddc624ab1ab395)) -- let me know if you're still having issues once you update.

**Comment by dotdotdotpaul** [comment]
Ah, no wonder, my install is still 1.9.x.  That explains a lot.  😄

**Comment by dotdotdotpaul** [comment]
Woot, works great, once I got the right version of Kayenta running.  😉

**Comment by maggieneterval** [comment]
Glad to hear, @dotdotdotpaul -- thanks again for testing this. Hope to meet you in person at the Spinnaker meetup next week!

**Pull request merged**